src/transformers/generation/utils.py

```diff
@@ -1234,6 +1234,17 @@ def generate(
                 "(https://huggingface.co/docs/transformers/main/en/main_classes/text_generation)"
             )
 
+        has_default_min_length = kwargs.get("min_length") is None and generation_config.min_length == 0
```

Comment:
```suggestion
        has_default_min_length = kwargs.get("min_length") is None and generation_config.min_length is not None
```
(some models can have a default value for min_length that is not 0)